### ISSUE TITLE
Add `no-arguments` rule

### ISSUE THREAD

**Issue by jfmengels** [jfmengels/eslint-plugin-fp#4]
This rule would forbid the use of the `arguments` variable implicitly given to the context of a function

### Fail

```js
function foo() {
  console.log(arguments);
}
```

### Valid

```js
function foo(a, b) {
  // anything not using arguments
}
```

I wonder whether the spread arguments should also be forbidden, probably either behind a flag or in an other rule (my current favorite option at the moment), as it does not work well with currying.

### Fail

```js
function foo(...args) {
}
```

### Valid

```js
function foo(args) { // passed as an array
}
```

Does anybody have any thoughts on this? Open to bikeshedding :)

**Comment by jfmengels** [comment]
First part fixed in 5e02f93. Will probably create a second rule for the forbidding `...args`.

**Comment by jfmengels** [comment]
`no-arguments` and `no-rest-parameters` are now available in master. Will release this as v2.1.0 soon.
